src/server/scripts/Spells/spell_holiday.cpp

```diff
@@ -42,75 +42,77 @@ enum SpellsPicnic
+        // If our player is no longer sit, remove all auras
+        if (target->GetStandState() != UNIT_STAND_STATE_SIT)
+        {
+            target->RemoveAurasDueToSpell(SPELL_ROMANTIC_PICNIC_ACHIEV);
+            target->RemoveAura(GetAura());
+            return;
+        }
+
+        target->CastSpell(target, SPELL_BASKET_CHECK); // unknown use, it targets Romantic Basket
+        target->CastSpell(target, RAND(SPELL_MEAL_EAT_VISUAL, SPELL_DRINK_VISUAL));
+
+        bool foundSomeone = false;
+        // For nearby players, check if they have the same aura. If so, cast Romantic Picnic (45123)
+        // required by achievement and "hearts" visual
+        std::list<Player*> playerList;
+        Trinity::AnyPlayerInObjectRangeCheck checker(target, INTERACTION_DISTANCE*2);
+        Trinity::PlayerListSearcher<Trinity::AnyPlayerInObjectRangeCheck> searcher(target, playerList, checker);
+        Cell::VisitWorldObjects(target, searcher, INTERACTION_DISTANCE * 2);
+        for (std::list<Player*>::const_iterator itr = playerList.begin(); itr != playerList.end(); ++itr)
+        {
+            if (foundSomeone)
+                continue;
 
-                target->CastSpell(target, SPELL_BASKET_CHECK, false); // unknown use, it targets Romantic Basket
-                target->CastSpell(target, RAND(SPELL_MEAL_EAT_VISUAL, SPELL_DRINK_VISUAL), false);
-
-                bool foundSomeone = false;
-                // For nearby players, check if they have the same aura. If so, cast Romantic Picnic (45123)
-                // required by achievement and "hearts" visual
-                std::list<Player*> playerList;
-                Trinity::AnyPlayerInObjectRangeCheck checker(target, INTERACTION_DISTANCE*2);
-                Trinity::PlayerListSearcher<Trinity::AnyPlayerInObjectRangeCheck> searcher(target, playerList, checker);
-                Cell::VisitWorldObjects(target, searcher, INTERACTION_DISTANCE * 2);
-                for (std::list<Player*>::const_iterator itr = playerList.begin(); itr != playerList.end(); ++itr)
+            if (Player* playerFound = (*itr))
+            {
+                if (target != playerFound && playerFound->HasAura(GetId()))
                 {
-                    if ((*itr) != target && (*itr)->HasAura(GetId())) // && (*itr)->GetStandState() == UNIT_STAND_STATE_SIT)
-                    {
-                        if (caster)
-                        {
-                            caster->CastSpell(*itr, SPELL_ROMANTIC_PICNIC_ACHIEV, true);
-                            caster->CastSpell(target, SPELL_ROMANTIC_PICNIC_ACHIEV, true);
-                        }
-                        foundSomeone = true;
-                        // break;
-                    }
+                    playerFound->CastSpell(playerFound, SPELL_ROMANTIC_PICNIC_ACHIEV, true);
+                    target->CastSpell(target, SPELL_ROMANTIC_PICNIC_ACHIEV, true);
+                    foundSomeone = true;
                 }
```

Comment:
or even better, how about adding a "break;" here after foundSomeone so we get out of the loop as soon as we are done ?

Reply:
ok, thx